packages/blockchain/src/index.ts

```diff
@@ -314,7 +312,7 @@ export default class Blockchain implements BlockchainInterface {
    * @hidden
    */
   _setCanonicalGenesisBlock(cb: any): void {
-    const genesisBlock = new Block(null, { common: this._common })
+    const genesisBlock = new Block(undefined, { common: this._common })
```

Comment:
Just for understanding: was this a necessary change which triggered some test failure or something before respectively came along with some code changes in `v3.0.0`? Or is this just an adjustment to make things cleaner?

Reply:
This first round of changes is to comply with the linter and `tsc`. There were several compilation failures and Blockchain code was not functional when bumping Block to 3.0.0.

Reply:
In this specific case, `null` is not informed as a valid type in the `constructor()`.

```typescript
  constructor(
    data: Buffer | [Buffer[], Buffer[], Buffer[]] | BlockData = {},
    opts: ChainOptions = {},
)
```

Reply:
@holgerd77 I am leaning towards allowing `null` in Block constructor. What are your thoughts? 

I can then revert the `null` to `undefined` changes made in this PR.

Reply:
Would also have a tendency to keep it "as is".

Reply:
got it. working on it.

Reply:
Right. In the end, I have changed my mind regarding this.
 
The TypeScript community in general is not much of a fan of `null` values representing `undefined` values. In fact, this is a broad issue on how `null` is much often misused.

So, as the default code from TypeScript version of Block wasn't allowing `null` in their signature, I stick to it, and for JavaScript consumers, though, I added [this check](https://github.com/ethereumjs/ethereumjs-vm/pull/682/files#diff-61ac76e0777f844b544ffa824f6ccbdfR32-R35), which makes sure the method will behave as expected, so if a JS code initializes Block with `data === null`,  `data` will become its [null-object](https://en.wikipedia.org/wiki/Null_object_pattern) equivalent with value `{}`.

The impact in our TS code of using `new Block(undefined, {})` is super low, only used in Block tests.